### PR TITLE
py-pydantic: add v2.7.4 with dep py-annotated-types

### PR DESCRIPTION
**Pull request by qwertos**
Depends on `py-annotated-types`, which is added as well.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pydantic/package.py [hunk 1]
```diff
@@ -14,16 +14,23 @@ class PyPydantic(PythonPackage):
...
+    depends_on("py-setuptools", type="build", when="@1")
+    depends_on("py-hatchling", type="build", when="@2")
+    depends_on("py-hatch-fancy-pypi-readme@22.5.0:", type="build", when="@2")
+    depends_on("py-typing-extensions@4.6.1:", when="@2.7.1:", type=("build", "run"))
     depends_on("py-typing-extensions@4.2:", when="@1.10.9:", type=("build", "run"))
     depends_on("py-typing-extensions@4.1:", when="@1.10:", type=("build", "run"))
     depends_on("py-typing-extensions@3.7.4.3:", type=("build", "run"))
 
-    depends_on("py-python-dotenv@0.10.4:", when="+dotenv", type=("build", "run"))
+    depends_on("py-annotated-types@0.4.0:", type=("build", "run"), when="@2.7.4:")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-pydantic/package.py:33]: Missing dependency.

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
Depends on `py-annotated-types`, which does not exit as recipe.

**Comment by wdconinc** [comment]
```
==> Installing py-annotated-types-0.7.0-7bnfgmww62j4e3huk2algxn2pjpiopjt [33/33]
==> No binary for py-annotated-types-0.7.0-7bnfgmww62j4e3huk2algxn2pjpiopjt found: installing from source
==> Fetching https://files.pythonhosted.org/packages/source/a/annotated_types/annotated_types-0.7.0.tar.gz
==> No patches needed for py-annotated-types
==> py-annotated-types: Executing phase: 'install'
==> py-annotated-types: Successfully installed py-annotated-types-0.7.0-7bnfgmww62j4e3huk2algxn2pjpiopjt
  Stage: 1.94s.  Install: 0.74s.  Post-install: 0.17s.  Total: 3.00s
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.3.0/py-annotated-types-0.7.0-7bnfgmww62j4e3huk2algxn2pjpiopjt
```
and
```
$ python 
Python 3.11.9 (main, Sep 13 2024, 13:12:09) [GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import annotated_types
>>> annotated_types.__version__
'0.7.0'
```

File: var/spack/repos/builtin/packages/py-pydantic/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-pydantic/package.py:33]: Added in 73a5353

**Review by bernhardkaindl — APPROVED** [review]
Build results on Ubuntu 22.04.4 LTS:
```py
bin/spack find --variants py-annotated-types@0.7.0%clang@15.0.7 py-annotated-types@0.7.0%clang@14.0.0 py-annotated-types@0.7.0%clang@14.0.0 py-annotated-types@0.7.0%gcc@9.5.0 py-annotated-types@0.7.0%gcc@8.4.0 py-annotated-types@0.7.0%gcc@13.1.0 py-annotated-types@0.7.0%gcc@12.3.0 py-annotated-types@0.7.0%gcc@11.4.0 py-annotated-types@0.7.0%gcc@11.4.0 py-annotated-types@0.7.0%gcc@10.5.0 py-pydantic@2.7.4%clang@15.0.7 py-pydantic@2.7.4%clang@14.0.0 py-pydantic@2.7.4%clang@14.0.0 py-pydantic@2.7.4%gcc@9.5.0 py-pydantic@2.7.4%gcc@8.4.0 py-pydantic@2.7.4%gcc@13.1.0 py-pydantic@2.7.4%gcc@12.3.0 py-pydantic@2.7.4%gcc@11.4.0 py-pydantic@2.7.4%gcc@11.4.0 py-pydantic@2.7.4%gcc@10.5.0
-- linux-ubuntu22.04-skylake / clang@14.0.0 ---------------------
py-annotated-types@0.7.0
py-pydantic@2.7.4~dotenv

-- linux-ubuntu22.04-skylake / clang@15.0.7 ---------------------
py-annotated-types@0.7.0
py-pydantic@2.7.4~dotenv

-- linux-ubuntu22.04-skylake / gcc@8.4.0 ------------------------
py-annotated-types@0.7.0
py-pydantic@2.7.4~dotenv

-- linux-ubuntu22.04-skylake / gcc@9.5.0 ------------------------
py-annotated-types@0.7.0
py-pydantic@2.7.4~dotenv

-- linux-ubuntu22.04-skylake / gcc@10.5.0 -----------------------
py-annotated-types@0.7.0
py-pydantic@2.7.4~dotenv

-- linux-ubuntu22.04-skylake / gcc@11.4.0 -----------------------
py-annotated-types@0.7.0
py-pydantic@2.7.4~dotenv

-- linux-ubuntu22.04-skylake / gcc@12.3.0 -----------------------
py-annotated-types@0.7.0
py-pydantic@2.7.4~dotenv

-- linux-ubuntu22.04-skylake / gcc@13.1.0 -----------------------
py-annotated-types@0.7.0
py-pydantic@2.7.4~dotenv
```
Generated by https://github.com/spack/build-quality-tools/blob/main/build_pr_changes.py
```py
build-quality-tools/build_pr_changes.py -may -c all
```

**Pull request merged**